server/oauth2.go

```diff
@@ -548,3 +552,68 @@ func validateRedirectURI(client storage.Client, redirectURI string) bool {
 	host, _, err := net.SplitHostPort(u.Host)
 	return err == nil && host == "localhost"
 }
+
+// connectorIDFromIDTokenHint tries to extract the connector used when the
+// passed ID token was issued. If anything goes wrong, it returns an empty
+// string, causing the connector selection to be presented.
+// Although it is not strictly required, this checks the token's signature.
+func (s *Server) connectorIDFromIDTokenHint(hint string) string {
+	tok, err := jwt.ParseSigned(hint)
+	if err != nil {
+		s.logger.Debugf("failed to parse JWT: %v", err)
+		return ""
+	}
+
+	// figure out key used for signing this
+	var usedKey string
+	for _, hdr := range tok.Headers {
+		if jose.SignatureAlgorithm(hdr.Algorithm) != jose.RS256 {
```

Comment:
In some distant future, I'd like to not hard code jose.RS256 everywhere. Maybe we can take this from a centrally configured place?